### PR TITLE
Add validations and model specs

### PR DESCRIPTION
**Pull request by Ewoskyx**
## Validations & Specs

In this PR I did;

**Add the following validations:**

**For the User model:**
* Name must not be blank.
* PostsCounter must be an integer greater than or equal to zero.

**For the Post model:**
* Title must not be blank.
* Title must not exceed 250 characters.
* CommentsCounter must be an integer greater than or equal to zero.
* LikesCounter must be an integer greater than or equal to zero.

**Add unit specs for all of your models' methods and validations.**

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'comment must be present' do
+      expect(Comment.all.length).to eq 2
+    end
+
+    it 'comment must have a correct text' do
+      expect(Comment.first.text).to eq 'Hi Tom!'
+    end
+  end
+end
```
Comment (medaminedev66) [inline spec/models/comment_spec.rb]: - [x] Following the project requirements, any custom methods in the models should have a corresponding unit test. Please add a unit test for the `update_comments_counter` method in the Comment model.

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    it 'like must be present' do
+      expect(Like.all.length).to eq 1
+    end
+
+    it 'like must have a correct author' do
+      expect(Like.first.author.name).to eq 'Tom'
+    end
+  end
```
Comment (medaminedev66) [inline spec/models/like_spec.rb:17]: - [x] As I mentioned in the previous comment, please ensure all the methods in the Like, User, and Post models have a corresponding unit test:
  - update_likes_counter (Like)
  - update_posts_counter (Post)
  - recent_comments (Post)
  - recent_posts (User)

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  describe 'Comment model' do
+    before(:each) do
+      first_user = User.create!(name: 'Tom', photo: 'photo.jpg', bio: 'Teacher from Mexico.')
+      first_post = Post.create!(author_id: first_user.id, title: 'Hello', text: 'This is my first post')
+      Comment.create!(post_id: first_post.id, author_id: first_user.id, text: 'Hi Tom!')
+    end
+
+    it 'comment must be present' do
+      expect(Comment.all.length).to eq 2
+    end
```
Comment (medaminedev66) [inline spec/models/comment_spec.rb]: - [x] This test fails after running `rspec spec`. Please, try to correct it.

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Ewoskyx   😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- `User` and `Post` models have validations.
- `User`, `Post`, `Comment` and `Like` models have **Validation** tests.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @Ewoskyx, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Linters ✔
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**